src/Journal.cpp

```diff
@@ -16,19 +16,486 @@
 #include "Journal.h"
 
 #include <algorithm>
+#include <functional>
+#include <map>
+#include <mutex>
+#include <optional>
+#include <type_traits>
 #include <unordered_map>
 #include <wx/app.h>
+#include <wx/event.h>
 #include <wx/filename.h>
+#include <wx/spinctrl.h>
+#include <wx/stattext.h>
+#include <wx/textctrl.h>
 #include <wx/textfile.h>
+#include <wx/tglbtn.h>
+#include <wx/toplevel.h>
+#include <wx/valgen.h>
 
-#include "MemoryX.h"
 #include "Prefs.h"
+#include "Theme.h"
+
+namespace Journal {
 
 namespace {
 
 constexpr auto SeparatorCharacter = ',';
+constexpr auto EscapeCharacter = '\\';
 constexpr auto CommentCharacter = '#';
 
+namespace Windows {
+
+/*!
+utilities to identify corresponding windows between recording and playback runs
+*/
+
+using Path = Identifier;
+constexpr auto PathSeparator = ':';
+
+/*! @return a predicate, on a pointer to a window,
+   that it has the given name.
+*/
+auto HasName( const wxString &name )
+{
+   return [name](auto *pWindow2){
+      return pWindow2
```

Comment:
I prefer the brevity of relying on the bool conversion.

I can write trailing && on this line, not the next.  I don't have strong feelings.  Putting it on the next is one of the silly rules of the MuseScore coding standard, and I don't know who I was trying to please with it.  But we don't use those rules.

auto in the function signature DOES help.  It allows a function to return a lambda.  I can't do it otherwise.  The alternative is changing return type to a std::function, with its overheads, and maybe defeating the inlining of std::find_if.

So I agree with only 1 out of 3 things.